test/lifecycle_test.dart

```diff
@@ -196,7 +196,8 @@ main() {
           'initialState': 'initial',
         };
 
-        final Map initialProps = unmodifiableMap({'getInitialState': (_) => initialState});
+        final Map initialProps =
+            unmodifiableMap({'getInitialState': (_) => initialState, 'initialState': (_) => initialState});
```

Comment:
Why is having both `getInitialState` and `initialState` as a key in this map necessary?

Reply:
because this test suite is shared it runs against component1 and component2 and because component2 does not have a getInitalState we need to provide both and only the one that it uses will be executed